File: xor/xor.go
package main

import "os"

const (
	XOR_KEY = 0xFF
	BUFFER_SIZE = 1024 * 2
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	path := "gta3.img"

	var (
		err error;
		src, dst *os.File;
		buff = make([]byte, BUFFER_SIZE);
		read_bytes int;
	)

	src, err = os.Open(path)
	check(err)
	defer src.Close()

	dst, err = os.Create(path + "-xor")
	check(err)
	defer dst.Close()

	for {
		read_bytes, err = src.Read(buff)
		if read_bytes == 0 {
			break
		}

		for i := 0; i < read_bytes; i++ {
			buff[i] ^= XOR_KEY
		}

		_, err = dst.Write(buff[:read_bytes])
		check(err)
	}
}

File: xor/xor.rs
use std::io::prelude::*;
use std::fs::File;

const XOR_KEY: u8 = 0xFF;
const BUFFER_SIZE: usize = 2048;

fn main() -> std::io::Result<()> {
    let mut buffer = [0; BUFFER_SIZE];

    let mut file_path = "gta3.img".to_owned();
    let mut input_file = File::open(&file_path)?;

    file_path.push_str("-xor");
    let mut output_file = File::create(&file_path)?;
   
    loop {
        let bytes_read = input_file.read(&mut buffer)?;
        if bytes_read == 0 {
            break;
        }

        for byte in buffer.iter_mut() {
            *byte ^= XOR_KEY;
        }

        output_file.write(&buffer[..bytes_read])?;
    }

    Ok(())
}